doc/rtd/tutorial/wsl.rst

```diff
@@ -0,0 +1,296 @@
+      path: /var/tmp/hello-world.txt
+      permissions: '0777'
+
+Save the file to ``%USERPROFILE%\.cloud-init\Ubuntu-Preview.user-data``.
+
+For example, if your username is ``me``, the path would be
+``C:\Users\me\.cloud-init\Ubuntu-Preview.user-data``.
+Ensure that the file is saved with the ``.user-data`` extension and
+not as a ``.txt`` file.
+
+What is user data?
+==================
+
+Before moving forward, let's inspect our :file:`user-data` file.
+
+We created the following contents:
+
+.. code-block:: yaml
+
+    #cloud-config
+    write_files:
+    - content: |
+        Hello from cloud-init
+      path: /var/tmp/hello-world.txt
+      permissions: '0770'
+
+The first line starts with ``#cloud-config``, which tells cloud-init
+what type of user data is in the config. Cloud-config is a YAML-based
+configuration type that tells cloud-init how to configure the instance
+being created. Multiple different format types are supported by
+cloud-init. For more information, see the
+:ref:`documentation describing different formats<user_data_formats>`.
+
+The remaining lines, as per
+:ref:`the Write Files module docs<mod-write_files>`, creates a file
+``/var/tmp/hello-world.txt`` with the content ``Hello from cloud-init`` and
+permissions allowing anybody on the system to read or write the file.
+
+Start the Ubuntu WSL instance
+=============================
+
+.. code-block:: doscon
+
+    PS> wsl --distribution Ubuntu-Preview
+
+The Ubuntu WSL instance will start, and you may be prompted for a username
+and password.
+
+.. code-block:: text
+
+    Installing, this may take a few minutes...
+    Please create a default UNIX user account. The username does not need to match your Windows username.
+    For more information visit: https://aka.ms/wslusers
+    Enter new UNIX username:
+    New password:
+    Retype new password:
+
+Once the credentials have been entered, you should see a welcome
+screen similar to the following:
+
+.. code-block:: text
+
+    Welcome to Ubuntu Noble Numbat (development branch) (GNU/Linux 5.15.146.1-microsoft-standard-WSL2 x86_64)
+
+    * Documentation:  https://help.ubuntu.com
+    * Management:     https://landscape.canonical.com
+    * Support:        https://ubuntu.com/pro
+
+    System information as of Mon Apr 22 21:06:49 UTC 2024
+
+    System load:  0.08                Processes:             51
+    Usage of /:   0.1% of 1006.85GB   Users logged in:       0
+    Memory usage: 4%                  IPv4 address for eth0: 172.29.240.255
+    Swap usage:   0%
+
+
+    This message is shown once a day. To disable it please create the
+    /root/.hushlogin file.
+    root@machine:/mnt/c/Users/me#
+
+You should now be in a shell inside the WSL instance.
+
+Verify that ``cloud-init`` ran successfully
+-------------------------------------------
+
+Before validating the user data, let's wait for ``cloud-init`` to complete
+successfully:
+
+.. code-block:: shell-session
+
+    $ cloud-init status --wait
```

Comment:
Do we want to mention cloud-id == wsl to show cloud-init detected the WSL datasource?